packages/back-end/src/api/openapi/openapi.yaml

```diff
@@ -13,7 +13,11 @@ info:
 
     We support both the HTTP Basic and Bearer authentication schemes for convenience.
 
-    You first need to generate a new Secret Key in GrowthBook by going to `Settings -> API Keys`.
+    You first need to generate a new API Key in GrowthBook by going to `Settings -> API Keys`. Different keys have different permissions:
+    
+    - **Secret keys**: These are sensitive and provide full access to the organization
+    - **User API keys**: These are sensitive and provide the same level of access as the user has to an organization
+    - **Read-only keys**: These are sensitive and provide read access to resources under an organization
```

Comment:
I think we should only have 2 types of API keys

1. **Personal Access Tokens** which have a `userId` and lookup the role in the organization.  Any user manage their own Personal Access Tokens.
2. **Secret Keys** which have an explicit `role` specified, but no `userId`.  Only Admins with the `manageApiKeys` permission can manage Secret Keys.

For existing secret API keys, they would fall under 2 above and would default to `role = "admin"` to match the existing behavior.  Admins can create new keys with `role = "readonly"` if they want (or any other role).

This approach will be less confusing to users, more flexible, and will have less hard-coded permission logic (e.g. if type = "read-only", throw an error).  With this change, every incoming request will have a role associated with it, so you can always just check if the role contains the permission.

Reply:
I've made changes around supporting a role property for ApiTokens. Please re-review.